pricegraph/src/orderbook/order.rs

```diff
@@ -1,15 +1,20 @@
 //! Data and logic related to token pair order management.
 
+#![allow(dead_code)]
+
+use super::UserMap;
 use crate::encoding::{Element, Price, TokenId, TokenPair, UserId};
+use crate::num;
 use std::cmp;
 use std::collections::HashMap;
 use std::f64;
 
-/// Type definition for a mapping of orders between buy and sell tokens.
+/// A type for collecting orders and building an order map that garantees that
+/// per-pair orders are sorted for optimal access.
 #[derive(Debug, Default)]
-pub struct OrderMap(HashMap<TokenId, HashMap<TokenId, Vec<Order>>>);
+pub struct OrderCollector(HashMap<TokenId, HashMap<TokenId, Vec<Order>>>);
 
-impl OrderMap {
+impl OrderCollector {
```

Comment:
it's like an OrderMapBuilder, isn't it? Any reason why you wanted this extra abstraction? So that the actual ordermap can be immutable?

Reply:
Its so nothing outside of the ordermap can modify the order of the per-pair orders and break the assumptions that they are always sorted.

Reply:
Its just a type for collecting the orders, and then once they are collected, then the order map can be built and the orders are guaranteed to stay sorted.